code/game/objects/structures/pit.dm

```diff
@@ -175,9 +175,12 @@
 		if(W.do_tool_interaction(TOOL_HATCHET, user, src, 3 SECONDS, "hacking away at", "hacking at"))
 			physically_destroyed(FALSE)
 		return TRUE
-	if(istype(W,/obj/item/pen))
+	if(IS_PEN(W))
 		var/msg = sanitize(input(user, "What should it say?", "Grave marker", message) as text|null)
-		if(msg)
+		if(!CanPhysicallyInteract(user))
```

Comment:
You could probably put this check in `can_use_tool()` on the tool archetype. Not blocking though.